npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/version_resolver.rb

```diff
@@ -436,6 +450,25 @@ def run_npm_checker(path:, version:)
           end
         end
 
+        def run_npm7_checker(version:)
+          SharedHelpers.run_shell_command(
+            "npm install #{version_install_arg(version: version)} --package-lock-only --dry-run=true --ignore-scripts"
+          )
+          nil
+        rescue SharedHelpers::HelperSubprocessFailed => e
+          raise if e.message.match?(NPM7_PEER_DEP_ERROR_REGEX)
+        end
+
+        def version_install_arg(version:)
+          source = dependency.requirements.find { |req| req[:source] && req[:source][:type] == "git" }
+
+          if source && source[:type] == "git"
+            "#{dependency.name}@#{source.url}##{version}"
```

Comment:
Won't always be a git requirement or nil? 

```suggestion
          git_req = dependency.requirements.find { |req| req[:source] && req[:source][:type] == "git" }

          if git_req
            "#{dependency.name}@#{git_req[:source][:url]}##{version}"
```

Reply:
Yeah, you're right 🎉